woocommerce/woocommerce-ios
# Add SIWA support

Fixes #2643 
Fixes #2654 

- [x] Make sure WordPressAuthenticator PR is merged: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/373

There are design changes in this PR now that we're also using the prologue screen from WPAuthenticator. We're updating the design of the empty store screen per p91TBi-38l-p2#comment-1534, so the design changes in this PR are likely just for release 4.9. We originally wanted to target this for 4.8, but there are two new strings and we've passed the translation window for 4.8 so Aaron thought it'd be nicer to just target `develop` for 4.9.

## Changes

- Enabled SIWA in project's "Signing & Capabilities" tab
- Updated `WordPressAuthenticator` pod to include [the changes to enable customizations of the prologue screens](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/373) in this PR
- In `AppDelegate`, in case the app presents the empty stores screen when it remembers the last login, added `storePickerCoordinator?.onDismiss` to display the authenticator login flow when the user taps to try with another account
- Added a new semantic color `authPrologueBottomBackgroundColor`, and updated the styles in `AuthenticationManager` for the prologue screens now that we're reusing the screens in the authenticator library. Our current design has the same colors for the bottom buttons container UI in both light and dark mode so we're passing the colors in light mode (previously, we forced the bottom buttons container UI to be light mode only). https://github.com/woocommerce/woocommerce-ios/blob/74d9f5b92b55fc9fa51336e4d72e1d3c0831bcf7/WooCommerce/Classes/Authentication/Prologue/LoginPrologueViewController.swift#L120-L123
- In `AuthenticationManager`, also passed in our `LoginPrologueViewController` to be the child view controller of the prologue top container view. Some more options are updated/added to support the prologue screens without sign up UX and with SIWA enabled. The code to display authentication was updated to show the sign in flow from WPAuthenticator library.
  - Sign up is a special case for SIWA, since the backend creates a WP.com account for the user if there's no WP.com account linked to the Apple ID yet. In this case, we sync the credentials and present the store picker screen (likely empty).
- Removed the Log In CTA at the bottom from `LoginPrologueViewController`, and updated the colors to use the new semantic color `authPrologueBottomBackgroundColor`
- Updated `EmptyStoresTableViewCell` - however, this cell will be removed when we implement the updated design soon (p91TBi-38l-p2#comment-1534) since it doesn't work in the new design anymore.
- Added `func restartAuthentication()` to `StorePickerViewControllerDelegate` when the user taps to try signing in with another account from the store picker screen (restarting the auth process). Implemented this in `StorePickerCoordinator` to dismiss the controller based on how it's presented in the first place in `showStorePicker`.
- Removed unused outlets/views in `StorePickerViewController` and its xib, and presented the Jetpack setup webview when the user taps on the "Connect the store with Jetpack" CTA
- Updated unit tests


## Testing

- Log out from the store
- Tap on the Log In button --> should see a sheet with 3 sign in methods: WP.com, Google, and Apple. There is also a link below to enter site address first. Make sure the event `login_prologue_jetpack_login_button_tapped` is fired in the console as well.
- Make sure each login method works, especially the Apple one

## Example screenshots

### Prologue screen

\ | light | dark
-- | -- | --
`develop` | ![Simulator Screen Shot - iPhone 8 Plus - 2020-08-12 at 18 38 18](https://user-images.githubusercontent.com/1945542/90006201-05ae4400-dccb-11ea-8dae-0b9173c6d714.png) | ![Simulator Screen Shot - iPhone 8 Plus - 2020-08-12 at 18 38 09](https://user-images.githubusercontent.com/1945542/90006189-02b35380-dccb-11ea-99cc-1bf3f805e55a.png)
PR branch | ![Simulator Screen Shot - iPhone 11 Pro - 2020-08-18 at 09 08 27](https://user-images.githubusercontent.com/1945542/90582274-e69a3f80-e1ff-11ea-9b68-f664b330d057.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-08-18 at 09 08 20](https://user-images.githubusercontent.com/1945542/90582271-e306b880-e1ff-11ea-961f-17f7150b6310.png)
PR branch - sign in sheet | ![Simulator Screen Shot - iPhone 11 Pro - 2020-08-18 at 09 08 29](https://user-images.githubusercontent.com/1945542/90582338-07fb2b80-e200-11ea-9458-4654e0cd6444.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-08-18 at 09 08 21](https://user-images.githubusercontent.com/1945542/90582326-05003b00-e200-11ea-9e23-5f8854cea7f4.png)

### Empty stores screen

light | dark
-- | --
![Simulator Screen Shot - iPhone 11 Pro - 2020-08-19 at 09 37 48](https://user-images.githubusercontent.com/1945542/90582200-bbafeb80-e1ff-11ea-8bd6-80bd2827a635.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-08-19 at 09 38 01](https://user-images.githubusercontent.com/1945542/90582202-bfdc0900-e1ff-11ea-81cf-df4f7e03ce9f.png)




Update release notes:
